sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -28,3 +29,45 @@ def get_token(self, *scopes, **kwargs):
     from collections import namedtuple
 
     AccessToken = namedtuple("AccessToken", ["token", "expires_on"])
+
+__all__ = ["AzureKeyCredential"]
+
+
+class AzureKeyCredential(object):
+    """Credential type used for authenticating to an Azure service.
+    It provides the ability to update the key without creating a new client.
+
+    :param str api_key: The API key to your Azure account.
+    :raises: TypeError
+    """
+
+    def __init__(self, api_key):
+        # type: (str) -> None
+        if not isinstance(api_key, six.string_types):
+            raise TypeError("api_key must be a string.")
+        self._api_key = api_key  # type: str
+
+    @property
+    def api_key(self):
```

Comment:
consider changing name to `key` as not all keys are API keys. Also, for consistency with other languages

Reply:
changed to `key`, thanks!

Reply:
This also applies to the documentation/comments you have in the PR